src/Bicep.Decompiler.IntegrationTests/DecompilationTests.cs

```diff
@@ -204,6 +204,9 @@ public void Decompiler_handles_strings_with_newlines(string newline, string esca
         [DataRow("greater(variables('a'), variables('b'))", "boolean", "a > b")]
         [DataRow("greaterOrEquals(variables('a'), variables('b'))", "boolean", "a >= b")]
         [DataRow("equals(variables('a'), variables('b'))", "boolean", "a == b")]
+        [DataRow("equals(toLower(variables('a')),toLower(variables('b')))", "boolean", "a =~ b")]
+        [DataRow("not(equals(variables('a'),variables('b')))","boolean", "a != b")]
+        [DataRow("not(equals(toLower(variables('a')),toLower(variables('b'))))", "boolean", "a !~ b")]
         public void Decompiler_handles_banned_function_replacement(string expression, string type, string expectedValue)
```

Comment:
It would be good to add an end-to-end test as well. You can do this by simply adding some outputs to this file: https://github.com/Azure/bicep/blob/main/src/Bicep.Decompiler.IntegrationTests/Working/expressions/main.json.

You can then regenerate the corresponding decompiled Bicep file by updating the test baselines: https://github.com/Azure/bicep/blob/main/CONTRIBUTING.md#updating-test-baselines.